serving_frontend.py

```diff
@@ -39,15 +42,198 @@ def predict_across_all_models(self, data, func):
 	def predict(self, model_name, data):
 		return self.service_manager.predict(model_name, data)
 
-	def get_endpoint_mapping(self):
-		return self.endpoint_mapping
+	def add_endpoint(self, api_definition, callback, **kwargs):
+		endpoint = api_definition.keys()[0]
+		method = api_definition[endpoint].keys()[0]
+		api_name = api_definition[endpoint][method]['operationId']
+		self.handler.add_endpoint(api_name, endpoint, partial(callback, **kwargs), [method.upper()])
+
+	def setup_openapi_endpoints(self):
+		models = self.service_manager.get_loaded_models()
+		# TODO: not hardcode host:port
+		self.openapi_endpoints = {
+			'swagger': '2.0',
+			'info': {
+				'version': '1.0.0',
+  				'title': 'Model Serving Apis'
+  			},
+  			'host': '127.0.0.1:5000',
+  			'schemes': ['http'],
+  			'paths': {},
+  		}
+
+
+		# 1. Predict endpoints
+		for model_name, model in models.iteritems():
+			# Get model signature
+			signature = model.signature()
```

Comment:
Corresponding to comments above, change to model.signature?

Reply:
Changed